packages/grafana-ui/src/components/GraphNG/nullInsertThreshold.ts

```diff
@@ -29,12 +29,16 @@ export function applyNullInsertThreshold(opts: NullInsertOptions): DataFrame {
     insertMode = INSERT_MODES.threshold;
   }
 
-  const refField = frame.fields.find((field) => {
-    // note: getFieldDisplayName() would require full DF[]
-    return refFieldName != null ? field.name === refFieldName : field.type === FieldType.time;
-  });
+  let refField: Field | undefined = undefined;
+  if (refFieldName != null) {
+    refField = frame.fields.find((field) => field.name === refFieldName);
+  } else {
+    refField =
+      frame.fields.find((field) => field.type === FieldType.time) ??
+      frame.fields.find((field) => field.type === FieldType.number);
+  }
```

Comment:
^ needed to update the above logic to match more closely what it was before due to failing test - we should only find a matching time or number field if a `refFieldName` is not provided...